### PR TITLE
fixed preapply commands and added postapply/postplan

### PR DESCRIPTION
**Pull request by mootpt**
Per #100 
- Fixed issue with preapply commands not being executed
- Added post commands
- Updated docs and examples

### CODE REVIEW

File: postrun/post_run.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Package postrun handles running commands after the
```
Comment (anubhavmishra) [inline postrun/post_run.go]: There is a lot of redundant code here with `prerun` package doing the same thing. We should restructure this so we can use the code for both `prerun` and `postrun` pacakges.

File: server/apply_executor.go [hunk 2]
```diff
@@ -178,6 +181,14 @@ func (a *ApplyExecutor) apply(ctx *CommandContext, repoDir string, plan models.P
 	}
 	ctx.Log.Info("apply succeeded")
 
+	// if there are post apply commands then run them
+	if len(config.PostApply.Commands) > 0 {
```
Comment (anubhavmishra) [inline server/apply_executor.go]: If we are adding `post_apply` might as well add `post_plan` too. Some people might want to use it.

Comment (mootpt) [inline server/apply_executor.go]: sure thing. for whatever reason i just figure `pre_apply` was the same thing as `post_plan`, but you are right, the steps are different. Let me add it.

File: postrun/post_run.go (hunk 1 above)
Comment (mootpt) [inline postrun/post_run.go]: Will do. I'll get rid of the copy paste replace.

Comment (mootpt) [inline postrun/post_run.go]: @anubhavmishra The code is basically the same, I could move duplicate functions with different names in a more general package "run" or we could just rename the package and functions to "run". What were you thinking?

Comment (anubhavmishra) [inline postrun/post_run.go]: yep! run might make more sense.

Comment (mootpt) [inline postrun/post_run.go]: okay @anubhavmishra , settled on a more general package and opted to pass in the stage name(i.e "pre_plan", "post_plan", etc)

File: server/apply_executor.go (hunk 2 above)
Comment (mootpt) [inline server/apply_executor.go]: ✅

**Comment by mootpt** [comment]
looks like e2e test failed. code seems fine. seems like environment variable doesn't exist in circle?

**Comment by anubhavmishra** [comment]
yep. we might need to change it so that end to end test only run on master builds.

**Comment by anubhavmishra** [comment]
https://github.com/hootsuite/atlantis/pull/108 should fix the e2e failures for now. We will come up with a better plan to figure out when and how to run e2e tests.

**Comment by anubhavmishra** [comment]
@mootpt https://github.com/hootsuite/atlantis/pull/108 is merged. you can rebase now.

**Comment by mootpt** [comment]
Sorry i missed this update. Thanks for the merge!

_Sent from my Samsung SM-G935T using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by anubhavmishra** [comment]
All good!

**Pull request merged**